src/index.ts

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+import { Background, Feature, Element, Rule } from 'gherkin-ast';
+import { PreCompiler } from 'gherking';
+import { ScenarioNumberingConfig } from './types';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const debug = require("debug")("gpc:scenario-numbering");
+
+const DEFAULT_CONFIG: ScenarioNumberingConfig = {
+    format: '${i}. ${name}'
+};
+
+class ScenarioNumbering implements PreCompiler {    
+    private config: ScenarioNumberingConfig;
+    private i: number = 0;
+    constructor(config?: ScenarioNumberingConfig) {
+        debug("Initializing ScenarioNumbering, config: %o", config)
+        this.config = {
+            ...DEFAULT_CONFIG,
+            ...(config || {}),
+        };
+    }
+
+    onFeature(feature: Feature) {
+        debug("Processing Scenarios of Feature: %s", feature.name)
+        feature.elements.forEach((element: Element | Rule) => {
+            if (!(element instanceof Background || element instanceof Rule)) {
+                element.name = this.config.format
```

Comment:
I'd suggest creating a function to this logic of processing the name (a string), so to avoid duplicate.